mypy/checkexpr.py

```diff
@@ -1450,6 +1501,15 @@ def check_call(
         """
         callee = get_proper_type(callee)
 
+        overloaded = self.is_generic_decorator_overload_call(callee, args)
```

Comment:
nit: check_call is pretty hot, should we move this inside `if isinstance(callee, CallableType):` on L1513?

Reply:
Yes, good idea!